torch/csrc/stable/tensor_struct.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <ATen/core/TensorAccessor.h>
```

Comment:
I believe the principle is we should not include from ATen in this file. 

As you point out in the PR description, the use of non-header only ArrayRef seems to be the main issue that prevents TensorAccessor from being header-only

From earlier discussion, I do believe we want to provide a lightweight TensorAccessor for convenience. Similar to my comment on the size/stride PR, let's hash this one out when Jane is back